perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet.hpp

```diff
@@ -0,0 +1,240 @@
+};
+
+class BEVDet
+{
+public:
+  BEVDet() {}
+  BEVDet(
+    const std::string & config_file, int n_img, std::vector<Eigen::Matrix3f> cams_intrin,
+    std::vector<Eigen::Quaternion<float>> cams2ego_rot,
+    std::vector<Eigen::Translation3f> cams2ego_trans, const std::string & onnx_file,
+    const std::string & engine_file);
+
+  int doInfer(
+    const camsData & cam_data, std::vector<Box> & out_detections, float & cost_time, int idx = -1);
+  ~BEVDet();
+
+protected:
+  void initParams(const std::string & config_file);
+
+  void initViewTransformer(
+    std::shared_ptr<int> & ranks_bev_ptr, std::shared_ptr<int> & ranks_depth_ptr,
+    std::shared_ptr<int> & ranks_feat_ptr, std::shared_ptr<int> & interval_starts_ptr,
+    std::shared_ptr<int> & interval_lengths_ptr);
+  void exportEngine(const std::string & onnxFile, const std::string & trtFile);
+  int initEngine(const std::string & engine_file);
+
+  int deserializeTRTEngine(const std::string & engine_file, nvinfer1::ICudaEngine ** engine_ptr);
+
+  void mallocDeviceMemory();
+
+  void initCamParams(
+    const std::vector<Eigen::Quaternion<float>> & curr_cams2ego_rot,
+    const std::vector<Eigen::Translation3f> & curr_cams2ego_trans,
+    const std::vector<Eigen::Matrix3f> & cams_intrin);
+
+  void getAdjBEVFeature(
+    const std::string & curr_scene_token, const Eigen::Quaternion<float> & ego2global_rot,
+    const Eigen::Translation3f & ego2global_trans);
+
+  void getCurr2AdjTransform(
+    const Eigen::Quaternion<float> & curr_ego2global_rot,
+    const Eigen::Quaternion<float> & adj_ego2global_rot,
+    const Eigen::Translation3f & curr_ego2global_trans,
+    const Eigen::Translation3f & adj_ego2global_trans, float * transform_dev);
+
+private:
+  int n_img_;
+
+  int src_img_h_;
+  int src_img_w_;
+  int input_img_h_;
+  int input_img_w_;
+  int crop_h_;
+  int crop_w_;
+  float resize_radio_;
+  int down_sample_;
+  int feat_h_;
+  int feat_w_;
+  int bev_h_;
+  int bev_w_;
+  int bevpool_channel_;
+
+  float depth_start_;
+  float depth_end_;
+  float depth_step_;
+  int depth_num_;
+
+  float x_start_;
+  float x_end_;
+  float x_step_;
+  int xgrid_num_;
+
+  float y_start_;
+  float y_end_;
+  float y_step_;
+  int ygrid_num_;
+
+  float z_start_;
+  float z_end_;
+  float z_step_;
+  int zgrid_num_;
+
+  std::vector<float> mean_;
+  std::vector<float> std_;
+
+  bool use_depth_;
+  bool use_adj_;
+  int adj_num_;
+
+  int class_num_;
+  float score_thresh_;
+  float nms_overlap_thresh_;
+  int nms_pre_maxnum_;
+  int nms_post_maxnum_;
+  std::vector<float> nms_rescale_factor_;
+  std::vector<int> class_num_pre_task_;
+  std::map<std::string, int> out_num_task_head_;
+
+  std::vector<Eigen::Matrix3f> cams_intrin_;
+  std::vector<Eigen::Quaternion<float>> cams2ego_rot_;
+  std::vector<Eigen::Translation3f> cams2ego_trans_;
+
+  Eigen::Matrix3f post_rot_;
+  Eigen::Translation3f post_trans_;
+
+  std::vector<size_t> trt_buffer_sizes_;
+  void ** trt_buffer_dev_;
```

Comment:
The `trt_buffer_dev_` member variable is a raw pointer.  Consider using `std::unique_ptr<void[]>` or `std::vector<void*>` to manage the memory automatically and prevent potential memory leaks. This enhances memory safety and reduces the burden on the programmer.